docs/articles/reference/upgrading-to-v16.md

```diff
@@ -0,0 +1,258 @@
+## Upgrading to v0.16
+
+Bytewax v0.16 introduces major changes to the way that custom Bytewax inputs are created, as well as improvements to and windowing and execution. In this article, we'll cover some of the updates we've made to our API to make upgrading to v0.16 easier.
+
+## Input changes
+
+In v0.16, we have restructured input to be based around partitions in order to support rescaling stateful dataflows in the future. We have also dropped the `Config` suffix to our input classes. As an example, `KafkaInputConfig` has been renamed to `KafkaInput`.
+
+`ManualInputConfig` is now replaced by two base superclasses that can be subclassed to create custom input sources. They are `DynamicInput` and `PartitionedInput`.
+
+### `DynamicInput`
+
+`DynamicInput` sources are input sources that support reading distinct items from any number of workers concurrently.
+
+Dynamic inputs must be able to supply disjoint data for each worker that is started. If multiple workers read the same data, the Dataflow will process those as duplicate items.
```

Comment:
This is true for all inputs, not just dynamic ones, and it something that is touched upon in the input docstrings. I'd just delete this line.